Black-Market/Black-Market/Scenes/SignUp/SignUpViewModel.swift

```diff
@@ -5,13 +5,51 @@
 //  Created by Junior Sancho on 7/30/23.
 //
 import SwiftUI
+import Combine
 
 class SignUpViewModel: ObservableObject {
     @Published var email = ""
-    @Published var password = ""
-    @Published var fullName = ""
+    @Published var newPassword = ""
+    @Published var confirmPassword = ""
+    
+    @Published var fetchError: NetworkError?
+    var isLoadingData = true
+    private let networkService: UserNetworkInterface
+    private var cancellable = Set<AnyCancellable>()
     
     var isFormValid: Bool {
-        email.isValidEmail && !fullName.isEmpty && password.isValidPassword
+        email.isValidEmail && newPassword.isValidPassword && confirmPassword.isValidPassword
+    }
+    
+    init(networkService: UserNetworkInterface = UserNetworkService()) {
+        self.networkService = networkService
+    }
+    
+    func signUp() {
+        let params = [
+            "email" : email,
+            "password1" : newPassword,
+            "password2" : confirmPassword
+        ]
+        networkService.signUp(params: params)
+            .receive(on: DispatchQueue.main)
+            .sink(receiveCompletion: { completion in
+                switch completion {
+                case .finished: break
+                case .failure(let error):
+                    if let error = error as? NetworkError {
```

Comment:
what happens otherwise? will the UI keep loading|nothing shows up|other?

Reply:
Done